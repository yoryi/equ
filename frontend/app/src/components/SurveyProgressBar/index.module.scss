:local(.container) {
  width: 100%;
  height: 4px;
  margin-top: 80px;
  background-color: var(--light-blue);

  :local(.progress) {
    height: 4px;
    background-color: var(--main-blue);
    border-top-right-radius: 2px;
    border-bottom-right-radius: 2px;
    transition: 0.15s width;
  }
}