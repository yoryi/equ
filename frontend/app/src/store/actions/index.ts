export * from "./admin"
export * from "./attachments"
export * from "./auth"
export * from "./follow"
export * from "./help"
export * from "./journeys"
export * from "./media"
export * from "./notifications"
export * from "./profile"
export * from "./searchAndFilters"
export * from "./settings"
export * from "./shared"
export * from "./student"
export * from "./surveys"
export * from "./university"
