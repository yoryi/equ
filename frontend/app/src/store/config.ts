interface Config {
  baseURL: string | null
}

export const config: Config = {
  baseURL: null,
}

export const initReduxStore = (baseURL: string) => {
  config.baseURL = baseURL
}
