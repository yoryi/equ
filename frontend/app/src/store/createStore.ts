// @ts-ignore
import { promiseMiddleware } from "@adobe/redux-saga-promise"
import { applyMiddleware, combineReducers,createStore } from "redux"
//Tools
import { composeWithDevTools } from "redux-devtools-extension"
import createSagaMiddleware from "redux-saga"
import { all } from "redux-saga/effects"

import admin from "./reducers/admin/reducer"
//Reducers
import auth from "./reducers/auth/reducer"
import follow from "./reducers/follow"
import media from "./reducers/media/reducer"
import notifications from "./reducers/notifications"
import profile from "./reducers/profile/reducer"
import searchAndFilters from "./reducers/searchAndFilters/reducer"
import settings from "./reducers/settings/reducer"
import shared from "./reducers/shared/reducer"
import student from "./reducers/student"
import surveys from "./reducers/surveys/reducer"
import university from "./reducers/university"
import { adminSaga } from "./sagas/admin/saga"
import { attachmentsSaga } from "./sagas/attachments/saga"
//Sagas
import { authSaga } from "./sagas/auth/saga"
import { followSaga } from "./sagas/follow"
import { helpSaga } from "./sagas/help"
import { journeySaga } from "./sagas/journeys/saga"
import { mediaSaga } from "./sagas/media/saga"
import { notificationsSaga } from "./sagas/notifications"
import { profileSaga } from "./sagas/profile/saga"
import { searchAndFiltersSaga } from "./sagas/searchAndFilters/saga"
import { settingsSaga } from "./sagas/settings/saga"
import { sharedSaga } from "./sagas/shared/saga"
import { studentSaga } from "./sagas/student"
import { surveysSaga } from "./sagas/surveys/saga"
import { universitySaga } from "./sagas/university"

const sagaMiddleware = createSagaMiddleware()

export const store = createStore(
  combineReducers({
    auth,
    media,
    notifications,
    profile,
    shared,
    settings,
    student,
    surveys,
    university,
    follow,
    searchAndFilters,
    admin,
  }),
  composeWithDevTools(applyMiddleware(promiseMiddleware, sagaMiddleware)),
)

function* saga() {
  yield all([
    ...authSaga,
    ...helpSaga,
    ...notificationsSaga,
    ...profileSaga,
    ...sharedSaga,
    ...settingsSaga,
    ...studentSaga,
    ...surveysSaga,
    ...journeySaga,
    ...attachmentsSaga,
    ...mediaSaga,
    ...universitySaga,
    ...followSaga,
    ...searchAndFiltersSaga,
    ...adminSaga,
  ])
}

sagaMiddleware.run(saga)
