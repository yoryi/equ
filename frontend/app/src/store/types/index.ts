export * from "./admin"
export * from "./auth"
export * from "./errors"
export * from "./ethnicity"
export * from "./follow"
export * from "./gender"
export * from "./hardship"
export * from "./help"
export * from "./highSchool"
export * from "./interest"
export * from "./notifications"
export * from "./profile"
export * from "./race"
export * from "./search"
export * from "./settings"
export * from "./shared"
export * from "./store"
export * from "./student"
export * from "./survey"
export * from "./university"
export * from "./utils"
