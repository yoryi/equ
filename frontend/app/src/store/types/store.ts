import { AdminReducer } from "../reducers/admin/reducer"
import { AuthReducer } from "../reducers/auth/reducer"
import { FollowReducer } from "../reducers/follow"
import { MediaReducer } from "../reducers/media/reducer"
import { NotificationsReducer } from "../reducers/notifications"
import { ProfileReducer } from "../reducers/profile/reducer"
import { SearchAndFiltersReducer } from "../reducers/searchAndFilters/reducer"
import { SettingsReducer } from "../reducers/settings/reducer"
import { SharedReducer } from "../reducers/shared/reducer"
import { StudentReducer } from "../reducers/student"
import { SurveysReducer } from "../reducers/surveys/reducer"
import { UniversityReducer } from "../reducers/university"

export interface ReduxState {
  auth: AuthReducer
  media: MediaReducer
  notifications: NotificationsReducer
  profile: ProfileReducer
  shared: SharedReducer
  settings: SettingsReducer
  student: StudentReducer
  surveys: SurveysReducer
  university: UniversityReducer
  follow: FollowReducer
  searchAndFilters: SearchAndFiltersReducer
  admin: AdminReducer
}
