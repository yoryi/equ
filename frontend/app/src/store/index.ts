import * as actions from "./actions"
import { initReduxStore } from "./config"
import { store } from "./createStore"

export { actions,initReduxStore, store }
