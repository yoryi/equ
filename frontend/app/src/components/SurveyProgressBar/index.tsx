//Utils
import _ from "lodash"
import * as React from "react"
//Hooks
import { useSelector } from "react-redux"
import { useLocation } from "react-router"

//Types
import { ReduxState } from "../../store/types"
//Styles
import Styles from "./index.module.scss"

const SurveyProgressBar: React.FC = () => {
  const id = parseInt(useLocation().pathname)

  const progress = useSelector<ReduxState, number>((state) =>
    state.surveys.survey
      ? _.findIndex(
          state.surveys.survey.questions,
          (question) => question.id === id,
        ) / state.surveys.survey.questions.length
      : 0,
  )

  return (
    <div className={Styles.container}>
      <div
        className={Styles.progress}
        style={{ width: `${progress * 100}%` }}
      />
    </div>
  )
}

export default SurveyProgressBar
