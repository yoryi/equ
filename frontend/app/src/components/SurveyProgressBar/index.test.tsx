//Testing
import { render } from "@testing-library/react"
import faker from "faker"
import _ from "lodash"
import * as React from "react"
//Redux store
import { Provider } from "react-redux"
//Navigation
import { MemoryRouter } from "react-router"
import configureStore from "redux-mock-store"
//Types
import { MockStore } from "redux-mock-store"

import { AnswerType,ReduxState } from "../../store/types"
//Components
import SurveyProgressBar from "./"

type TestStore = Pick<ReduxState, "surveys">

describe(`SurveyProgressBar component`, () => {
  faker.seed(1000)

  const mockStore = configureStore<TestStore>()
  let store: MockStore<TestStore>
  beforeEach(() => {
    store = mockStore({
      surveys: {
        surveys: [],
        isSurveyLoading: false,
        survey: {
          id: faker.random.number(10000),
          title: faker.lorem.words(2),
          description: faker.lorem.sentence(),
          questions: [
            {
              id: 1,
              title: faker.lorem.sentence(),
              studentPreferenceType: faker.random.number(10),
              answerType: AnswerType.SingleChoice,
              answer: undefined,
              body: _.times(4, (i) => ({
                key: i,
                content: faker.lorem.words(2),
              })),
            },
          ],
        },
      },
    })
  })

  const wrapper: React.FC = ({ children }) => (
    <MemoryRouter initialEntries={[`1`]}>
      <Provider store={store}>{children}</Provider>
    </MemoryRouter>
  )

  describe(`should render and match the snapshot`, () => {
    it(`survey is loaded`, () => {
      const { container } = render(<SurveyProgressBar />, { wrapper })

      expect(container.firstChild).toMatchSnapshot()
    })

    it(`survey is not loaded`, () => {
      store = mockStore({
        surveys: {
          surveys: [],
          isSurveyLoading: false,
          survey: null,
        },
      })

      const { container } = render(<SurveyProgressBar />, { wrapper })

      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
