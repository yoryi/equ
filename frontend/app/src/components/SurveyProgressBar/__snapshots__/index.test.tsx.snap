// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyProgressBar component should render and match the snapshot survey is loaded 1`] = `
<div
  class="container"
>
  <div
    class="progress"
    style="width: 0%;"
  />
</div>
`;

exports[`SurveyProgressBar component should render and match the snapshot survey is not loaded 1`] = `
<div
  class="container"
>
  <div
    class="progress"
    style="width: 0%;"
  />
</div>
`;
